envoy/network/connection.h

```diff
@@ -151,6 +175,15 @@ class Connection : public Event::DeferredDeletable,
    */
   virtual bool isHalfCloseEnabled() const PURE;
 
+  /**
+   * Enables or disables connection closure management via the filter manager.
+   * When enabled, the connection will verify the closure status of both read and write filters
+   * before completing the close operation.
+   *
+   * @param enabled True to enable closure via the filter manager; false is the default.
+   */
+  virtual void enableCloseThroughFilterManager(bool enabled) PURE;
```

Comment:
I don't think this is necessary because the feature will only be used when some filter call the disableClose(true) or return `StopIterationAndDontClose`. Additional switch make no much sense.

Reply:
I kind of leaning forward to keep this, but if you have a strong opinion we can remove. I treat this as a red button to enable the close by checking the aggregated filter's status, which is managed by the filter manager. When it is enabled it will actually check the filter status before closing, and if is disabled it will be the normal case even the filter manager thinks we cannot close.

Reply:
Done, and I marked as default false when I am baking this feature.